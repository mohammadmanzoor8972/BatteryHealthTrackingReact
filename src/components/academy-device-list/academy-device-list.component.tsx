import * as React from 'react'
import * as styles from './user-hobbies-list.scss'
import { Academy } from '../../models/academy'
import { Battery } from '../../models/battery';
import calculateBattery from '../../utilities/batteryLevel';

export interface AcademyDeviceListProps {
  academys: Academy
  batteries: Battery[]
}

interface NoBattriesPlaceholderProps {
  academys: Academy
}

const NoBattriesPlaceholder = ({academys}: NoBattriesPlaceholderProps) =>
  <span>{`Press add button to add hobby!`}</span>



export const AcademyDeviceList = ({academys, batteries}: AcademyDeviceListProps) => {
  


  return (<div className={styles.container}>
    {
      batteries && Object.keys(batteries).map(batery => (
        <div key={`${batery}`} className={styles.item}>
          {batery} 
          <span> {calculateBattery(batery, batteries)}%</span>
          {calculateBattery(batery, batteries) >30 ? <span style={{color:'red'}}>Change</span> : null}
     
        </div>
      ))
    }
  </div>)
}


