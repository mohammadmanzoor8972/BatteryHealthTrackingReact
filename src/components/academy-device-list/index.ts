export * from './academy-device-list.component'
