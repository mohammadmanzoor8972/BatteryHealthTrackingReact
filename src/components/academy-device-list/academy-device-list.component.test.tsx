import * as React from 'react'
import { mount } from 'enzyme'
import { AcademyDeviceList, calculateBattery } from './academy-device-list.component'


describe('AcademyDeviceList component', () => {
  const mockUser: any = {
    academyId: 30007,
    batteries: {}
  }

  it('should render no devics placeholder when no hobbies defined', () => {
    expect(
      mount(<AcademyDeviceList academys={mockUser} batteries={mockUser.batteries}  />)
    ).toMatchSnapshot()
  })

  it('should render devices', () => {
    expect(
      mount(<AcademyDeviceList academys={mockUser} batteries={[]}/>)
    ).toMatchSnapshot()
  })

  it('should should calculate', () => {
    let calculate = calculateBattery("TT-C67ML-A-0001-02585",{'TT-C67ML-A-0001-02585':[{"academyId":30007,"batteryLevel":0.47,"employeeId":"T1001061","serialNumber":"TT-C67ML-A-0001-02585","timestamp":"2019-05-17T07:39:57.884+01:00"},
    {"academyId":30007,"batteryLevel":0.38,"employeeId":"T1001061","serialNumber":"TT-C67ML-A-0001-02585","timestamp":"2019-05-17T15:09:58.221+01:00"}]})
    console.log(calculate)
    expect(
      calculate).toEqual('8');
  })



})