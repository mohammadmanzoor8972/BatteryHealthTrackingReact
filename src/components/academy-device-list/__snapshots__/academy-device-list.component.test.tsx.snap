// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademyDeviceList component should render devices 1`] = `
<Component
  academys={
    Object {
      "academyId": 30007,
      "batteries": Object {},
    }
  }
  batteries={Array []}
>
  <div
    className="container"
  />
</Component>
`;

exports[`AcademyDeviceList component should render no devics placeholder when no hobbies defined 1`] = `
<Component
  academys={
    Object {
      "academyId": 30007,
      "batteries": Object {},
    }
  }
  batteries={Object {}}
>
  <div
    className="container"
  />
</Component>
`;
